decidim-core/lib/decidim/filter_form_builder.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+# rubocop:disable Metrics/ParameterLists
+require "decidim/form_builder"
+
+module Decidim
+  # This custom FormBuilder is used to create resource filter forms
+  class FilterFormBuilder < FormBuilder
+    # Wrap the radio buttons collection in a custom fieldset.
+    # It also renders the inputs inside its labels.
+    def collection_radio_buttons(method, collection, value_method, label_method, options = {}, html_options = {})
+      fieldset_wrapper options[:legend_title] do
+        super(method, collection, value_method, label_method, options, html_options) do |b|
+          if block_given?
+            yield b
```

Comment:
It's not used because I'm not using the block in my filters but I thought it is a good practise to handle that edge case (since I'm overriding a form builder method).